tests/cpp17/generated_cpp17/optional_scalars_generated.h

```diff
@@ -19,6 +19,8 @@ struct ScalarStuff;
 struct ScalarStuffBuilder;
 struct ScalarStuffT;
 
+template<typename T> struct EnumTraits;
```

Comment:
what happens for a FlatBuffers schema that doesn't use a namespace? Are dumping this and the functions below in the global namespace?

I wonder if it makes sense to somehow always have them under `flatbuffers::`.. then the above decl can even be moved to our headers?